fog/ledger/server/src/db_fetcher.rs

```diff
@@ -144,11 +179,16 @@ impl<DB: Ledger, E: LedgerEnclaveProxy + Clone + Send + Sync + 'static> DbFetche
             // invocation. We want to keep loading blocks as long as we have data to load,
             // but that could take some time which is why the loop is also gated
             // on the stop trigger in case a stop is requested during loading.
-            while self.load_block_data() && !self.stop_requested.load(Ordering::SeqCst) {
+            while next_block_index < block_range.end_block
```

Comment:
nit: I try to avoid doing direct comparisons of `BlockRange` fields in calling code. Consider using `overlaps` (which I think works here because you checked that `next_block_index` is greater than `start_index`)